python/lsst/afw/display/interface.py

```diff
@@ -230,17 +257,22 @@ def __str__(self):
     # Handle Displays, including the default one (the frame to use when a user specifies None)
     #
     @staticmethod
-    def setDefaultBackend(backend):
+    def setDefaultBackend(backend, *args, **kwargs):
+        Display._args = args
+        Display._kwargs = kwargs
         try:
-            _makeDisplayImpl(None, backend)
+            _makeDisplayImpl(None, backend, *Display._args, **Display._kwargs)
         except Exception as e:
             raise RuntimeError("Unable to set backend to %s: \"%s\"" % (backend, e))
 
         Display._defaultBackend = backend
 
     @staticmethod
-    def getDefaultBackend():
-        return Display._defaultBackend
+    def getDefaultBackend(optional_args=False):
```

Comment:
Thanks for adding `optional_args` here.

I'm still worried about this interface, though. In particular, the fact that you can't in general do `afwDisplay.setDefaultBackend(afwDisplay.getDefaultBackend())` seems like a red flag.

I'm wondering if we actually need to adjust how the information about the backend is being encapsulated. Rather than a naked string and some free-floating arguments we need to remember to associate with it, you could bundle it all up into a tiny class:

```python
class DisplayBackend(object):
    @property
    def name(self):
        return (self.__class__.__name__[0].lower() +
                self.__class__.__name__[1:])

class VirtualDevice(DisplayBackend):
    pass

class Firefly(DisplayBackend):
    def __init__(self, host="localhost", port="8080"):
        self.host, self.port = host, port
```

The backend itself then tells you all you need to know about it, and you no longer have to worry about keeping your `args` in sync. This also makes handling arguments in `Display.__init__` easier: you no longer need to worry about the case when `self._args` or `self._kwargs` doesn't exist. In fact, you can just drop the `*args, **kwargs` arguments to `Display.__init__` entirely, and just have the user pass `DisplayBackend` subclasses with all the arguments already provided.

I should add that this isn't a requirement. In fact, I'd need to think harder about whether it's even a good idea: a new class makes life harder for backend implementers, you'd need to rework `_makeDisplayImpl` to handle it, and, logically, I'm not sure whether information about things like hosts and ports belongs with the backend at all (is the host it's running on a property of the backend or a property of the `Display`?).

So I'm not sure how to proceed here, but the current design does give me the cold shivers.